vault/core.go

```diff
@@ -1405,11 +1452,6 @@ func (c *Core) unsealInternal(ctx context.Context, masterKey []byte) (bool, erro
 		go c.runStandby(c.standbyDoneCh, c.manualStepDownCh, c.standbyStopCh)
 	}
 
-	// Force a cache bust here, which will also run migration code
-	if c.seal.RecoveryKeySupported() {
-		c.seal.SetRecoveryConfig(ctx, nil)
-	}
```

Comment:
Do we fully understand the repercussions of removing this? I'm curious about this comment and what migration code we are no longer running?

Reply:
This was a very hard decision I took. I have given it a lot of brain time and have felt that this is okay. That comment there was never understood though.

Reply:
I suspect it's talking about this migration, which we are no longer running. https://github.com/hashicorp/vault/blob/ba76c080ad1f0254ab6e9eb4671c312f7b5dbe67/vault/seal_autoseal.go#L324-L326 

Seems concerning to me, I think we need to continue to ensure that's run.

Reply:
You only added back the non-HA case. I suspect this is challenging because it removes the work that adjustCoreForMigration does?

Reply:
Okay. Alright. But that doesn't seem like it will have direct impact on the seal migration though. It seems to me that it can safely be done after migration is done. I've reintroduced this block for the non-ha case. For the ha, post seal migration, we are anyways busting the cache. Do you also think that this will address it?

Reply:
Yes. I've added only for the non-ha case since the ha case does regardless. Yes, it it undoes the `adjustCoreForMigration`.

Reply:
Is it being called in the standard HA when a seal migration is not going to be run?

Reply:
Yes. `migrateSeal` will be a no-op and returns nil. The flow will continue and reset the recovery config. (https://github.com/hashicorp/vault/pull/8103/files#diff-c2ee7e27b76414e630dfea47791364b0L470)

Reply:
@vishalnayak that link didn't work

Reply:
https://github.com/hashicorp/vault/blob/master/vault/ha.go#L471

Reply:
https://github.com/hashicorp/vault/blob/raft-seal-migration/vault/ha.go#L483